test/rubygems/test_gem_safe_yaml.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative "helper"
+
+Gem.load_yaml
+
+class TestGemSafeYAML < Gem::TestCase
+  def test_aliases_enabled_by_default
+    assert_predicate Gem::SafeYAML, :aliases_enabled?
+    assert_equal({ "a" => "a", "b" => "a" }, Gem::SafeYAML.safe_load("a: &a a\nb: *a\n"))
+  end
+
+  def test_aliases_disabled
+    aliases_enabled = Gem::SafeYAML.aliases_enabled?
+    Gem::SafeYAML.aliases_enabled = false
+    refute_predicate Gem::SafeYAML, :aliases_enabled?
+    assert_raise Psych::AliasesNotEnabled do
```

Comment:
Seems `Psych::AliasesNotEnabled` is not available in Psych distributed with Ruby bellow 3.2. What about to check for `BadAlias`? That should be compatible.

https://github.com/ruby/psych/blob/be0ba74e5613c20f213403e15914d24944c2652d/lib/psych/exception.rb#L9-L10